doc/extractor_userdocs.py

```diff
@@ -0,0 +1,270 @@
+# -*- coding: utf-8 -*-
+#
+# conf.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+from pprint import pformat
+import os
+import glob
+import json
+from itertools import chain, combinations
+import logging
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger()
+
+
+def relative_glob(*pattern, basedir=os.curdir, **kwargs):
+    tobase = os.path.relpath(basedir, os.curdir)
+    tohere = os.path.relpath(os.curdir, basedir)
+    # prefix all patterns with basedir and expand
+    names = chain(*[glob.glob(os.path.join(tobase, pat), **kwargs) for pat in pattern])
+    # remove prefix from all expanded names
+    return [name[len(tobase)+1:] for name in names]
+
+
+def UserDocExtractor(
+        filenames,
+        basedir="..",
+        replace_ext='.rst',
+        outdir="from_cpp/"
+        ):
+    """
+    Extract all user documentation from given files.
+
+    This method searches for "BeginUserDocs" and "EndUserDocs" keywords and
+    extracts all text inbetween as user-level documentation. The keyword
+    "BeginUserDocs" may optionally be followed by a colon ":" and a comma
+    separated list of tags till the end of the line.
```

Comment:
Maybe mention that tags may contain spaces, as that is not commonly supported elsewhere and people might use unneeded underscores if they don't know about the feature.

Reply:
done